daq/combine_reports_from_date_range.py

```diff
@@ -0,0 +1,106 @@
+import os
+import logger 
+import sys
+import datetime
+import re
+import json
+import functools
+
+import configurator
+from gcp import GcpManager
+from report import MdTable
+
+LOGGER = logger.get_logger('combine_reports_from_date_range')
+
+def _render_results(results):
+    def reduce_labels(sets):
+        return sorted(functools.reduce(lambda cur, acc: acc.union(cur), sets, set())) 
+    result_label_sets = map(lambda test: set(test.keys()), results['tests'].values())
+    result_labels = reduce_labels(result_label_sets)
+    tests_table = MdTable(['test'] + result_labels)
+    if results['tests']:
+        for test, result in results['tests'].items():
+            tests_table.add_row([test, *[str(result.get(label, 0)) for label in result_labels]]) 
+    else:
+        tests_table.add_row(['No results'])
+
+    category_label_sets = map(lambda category: set(category.keys()), results['categories'].values())
+    result_labels = reduce_labels(category_label_sets)
+    categories_table = MdTable(['categories'] + result_labels)
+    if results['categories']:
+        for category, result in results['categories'].items():
+            categories_table.add_row([category, *[str(result.get(label, 0)) for label in result_labels]]) 
+    else:
+        categories_table.add_row(['No results'])
+    
+    missing_table = MdTable(['missing tests', 'count'])
+    if results['missing']:
+        for test in sorted(results['missing'].keys()):
+            missing_table.add_row([test, str(results['missing'][test])]) 
+    else:
+        missing_table.add_row(['None'])
+    
+    return "\n%s\n" * 3 % (tests_table.render(), categories_table.render(), missing_table.render())    
+
+def main(device, start=None, end=None, gcp=None, reports_dir=os.path.join('inst', 'reports')):
+    aggregate = {'tests': {}, 'categories': {}, 'missing': {}}
+    if gcp:
```

Comment:
I'm not sure if this is right because it will change the behavior in somewhat unexpected ways.  E.g., the use of the tool should be to mine the local info for something. The behavior of the gcp_cred variable is "and then also do things with the cloud" -- not instead of.  Could make it a command-line option to pull from cloud.  Ideally for the cloud variant it would be a cloud function or similar that's triggered somehow through a cloud-operation... I don't think mixing "local tool with cloud data" is quite the right operating model.

Reply:
But the  tool is not intended to be just a “local tool”, it’s intended to be a script to perform an action which is to combine reports regardless of where the reports are. If GCP is defined, it makes sense to combine reports stored there instead of partial information stored locally.